perma_web/perma/views/service.py

```diff
@@ -1,37 +1,95 @@
 import json, logging, pytz
+from werkzeug.security import safe_str_cmp
 
 from django.core import serializers
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseForbidden
 from django.core.urlresolvers import reverse
 from django.shortcuts import redirect
 from django.utils import timezone
+from django.conf import settings
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+from django.views.decorators.debug import sensitive_post_parameters
 
 from perma.models import WeekStats, MinuteStats
-from perma.utils import json_serial, send_user_email
+from perma.utils import json_serial
+from perma.email import sync_cm_list, send_user_email, \
+     send_admin_email, registrar_users_plus_stats \
+     # send_user_bulk_email
 
-logger = logging.getLogger(__name__)
-
-
-def email_confirm(request):
-    """
-    A service that sends a message to a user about a perma link.
-    """
-
-    email_address = request.POST.get('email_address')
-    link_url = request.POST.get('link_url')
 
-    if not email_address or not link_url:
-        return HttpResponse(status=400)
-
-    send_user_email(
-        "The Perma link you requested",
-        "%s \n\n(This link is the Perma link)" % link_url,
-        email_address
-    )
+logger = logging.getLogger(__name__)
 
-    response_object = {"sent": True}
+@csrf_exempt
+@sensitive_post_parameters()
+@require_http_methods(["POST"])
+def cm_sync(request):
+    '''
+       Sync our current list of registrar users plus some associated metadata
+       to Campaign Monitor.
+    '''
 
-    return HttpResponse(json.dumps(response_object), content_type="application/json", status=200)
+    # Use something like this:
+    # curl -i -X POST --data "key=<INTERNAL_SERVICES_KEY>" http://perma.dev:8000/service/cm-sync/
+    if safe_str_cmp(request.POST.get('key',""), settings.INTERNAL_SERVICES_KEY):
+        reports = sync_cm_list(settings.CAMPAIGN_MONITOR_REGISTRAR_LIST,
+                               registrar_users_plus_stats(destination='cm'))
+        if reports["import"]["duplicates_in_import_list"]:
+            logger.error("Duplicate reigstrar users sent to Campaign Monitor. Check sync logic.")
+        send_admin_email("Registrar Users Synced to Campaign Monitor",
+                          settings.DEFAULT_FROM_EMAIL,
+                          request,
+                          'email/admin/sync_to_cm.txt',
+                          {"reports": reports})
+        return HttpResponse(json.dumps(reports), content_type="application/json", status=200)
+    else:
+        return HttpResponseForbidden("<h1>Forbidden</h1>")
+
+@csrf_exempt
+@sensitive_post_parameters()
+@require_http_methods(["POST"])
+def ping_registrar_users(request):
```

Comment:
As we discussed, if this is being run manually by devs, it might make sense as a fab task.

If it's being run manually by non-dev admins, it could instead be added as a button in the Django admin: http://stackoverflow.com/a/24553887/307769